### PR TITLE
Add an EVP demo for AES key wrap

### PR DESCRIPTION
**Pull request by faramir-dev**
Fixes openssl#14119

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

**Review by hlandau — APPROVED** [review]

**Comment by t8m** [comment]
Have you tested this? It looks like there is missing `EVP_CIPHER_CTX_set_flags(ctx, EVP_CIPHER_CTX_FLAG_WRAP_ALLOW);` call. However I do not see any check for this flag in provider based implementation so unless the implementation is legacy, setting the flag is apparently unnecessary.

**Comment by t8m** [comment]
As the flag was mostly _I really know what I am doing_ measure it is probably OK that we ignore it in the provider based implementation.

File: demos/cipher/aeskeywrap.c [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+const char *propq = NULL;
+
+int aes_wrap_encrypt(void)
+{
+    int ret = 0;
+    EVP_CIPHER_CTX *ctx;
+    EVP_CIPHER *cipher = NULL;
+    int outlen, tmplen;
+    unsigned char outbuf[1024];
+    unsigned char outtag[16];
```
Comment (slontis) [inline demos/cipher/aeskeywrap.c:63]: outtag is unused?

File: demos/cipher/aeskeywrap.c [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+    return ret;
+}
+
+int aes_wrap_decrypt(void)
+{
+    int ret = 0;
+    EVP_CIPHER_CTX *ctx;
+    EVP_CIPHER *cipher = NULL;
+    int outlen, tmplen, rv;
```
Comment (slontis) [inline demos/cipher/aeskeywrap.c:118]: rv is unused

**Review by t8m — APPROVED** [review]

**Review by slontis — APPROVED** [review]

**Comment by paulidale** [comment]
+1 good.

**Comment by hlandau** [comment]
Still OK.

**Comment by paulidale** [comment]
and merged...

**Pull request closed without merging**